src/app/search-page/search-page.controller.js

```diff
@@ -1,11 +1,26 @@
 export default class SearchPageController {
-    constructor(latestSearchesUtil) {
+    constructor(geoNames, latestSearchesUtil, $timeout, $state) {
+        this.geoNames = geoNames;
         this.latestSearchesUtil = latestSearchesUtil;
+        this.$timeout = $timeout;
+        this.$state = $state;
+        this.nearbyCities = [];
     }
 
     $onInit() {
+        this.cityParam = this.$state.params.city.toLowerCase();
```

Comment:
check if city param exists, because if enter directly to /search, app crashs and doesn't show the recent cities list